org.eclipse.jdt.debug.ui/ui/org/eclipse/jdt/internal/debug/ui/ObjectComparison.java

```diff
@@ -0,0 +1,1067 @@
+		int i;
+		StringBuilder contentBuilder = new StringBuilder();
+		String val;
+		for (IVariable v : value.getVariables()) {
+			if (v.getName().equalsIgnoreCase("value")) {
+				for (IVariable inner : v.getValue().getVariables()) {
+					val = inner.getValue().getValueString();
+					i = Integer.parseInt(val);
+					char[] chars = Character.toChars(i);
+					for (char e : chars) {
+						contentBuilder.append(e);
+					}
+				}
+			}
+		}
+		return contentBuilder.toString();
+	}
+
+	/**
+	 * Parent method for extracting set contents for given collection of selected objects
+	 *
+	 * @param selections
+	 *            List of IStructuredSelection
+	 * @throws Exception
+	 * @return Returns a Map containing selected IJavaVariable and its extracted Set contents
+	 */
+	public Map<IJavaVariable, Object> setExtraction(List<IStructuredSelection> selections) throws Exception {
+		List<String> contents;
+		String message;
+		Map<IJavaVariable, Object> result = new HashMap<>();
+		for (Object selection : selections) {
+			if (selection instanceof IJavaVariable selectedObject) {
+				if (selectedObject.getValue() instanceof IJavaObject javaObject) {
+					contents = setElementsExtraction(javaObject);
+					if (contents != null) {
+						result.put(selectedObject, contents);
+					} else {
+						message = NLS.bind(DebugUIMessages.CompareObjectsFailedException, new Object[] { selectedObject });
+						throw new Exception(message);
+					}
+				}
+			}
+		}
+		return result;
+	}
+
+	/**
+	 * Extract Set contents for the given IJavaObject tyoe
+	 *
+	 * @param javaObject1
+	 *            Selected Set object
+	 * @throws DebugException
+	 * @return Returns a List of Set contents
+	 */
+	@SuppressWarnings("nls")
+	public List<String> setElementsExtraction(IJavaObject javaObject1) throws DebugException {
+
+		IJavaFieldVariable root1 = javaObject1.getField("map", false);
+		if (root1 == null && javaObject1.getReferenceTypeName().startsWith("java.util.TreeSet")) {
+			IJavaFieldVariable map1 = javaObject1.getField("m", false);
+			if (map1 != null) {
+				Map<String, Object> result;
+				result = extractFromTreeMaps((IJavaObject) map1.getValue());
+				@SuppressWarnings("unchecked")
+				List<String> l1 = (List<String>) result.get(KEYSET_1);
+				return l1;
+			}
+		}
+		if (root1 == null && javaObject1.getReferenceTypeName().startsWith("java.util.concurrent.CopyOnWriteArraySet")) {
+			IJavaFieldVariable map1 = javaObject1.getField("al", false);
+			if (map1 != null) {
+				return extractFromConcurrentSets((IJavaObject) map1.getValue());
+			}
+		}
+		if (root1 == null) {
+			return null;
+		}
+		IJavaObject map1 = (IJavaObject) root1.getValue();
+		IJavaFieldVariable elements1 = map1.getField("table", false);
+		List<String> l1 = new ArrayList<>();
+		Object value;
+		if (elements1.getValue() != null) {
+			if (elements1.getValue() instanceof IJavaArray arr) {
+				if (arr != null) {
```

Comment:
an "instanceof" IJavaArray is always != null, as null is not an instanceof anything